app/controllers/registration_controller.rb

```diff
@@ -0,0 +1,36 @@
+class RegistrationController < ApplicationController
+  # before_action :confirm_password, only: [:create, :update]
+
+  def create
+    # debugger
+    @user =  User.new(user_params)
+    if @user.save
+      VerificationMailer.verify(@user, verificationUrl).deliver
+      flash[:notice] = "User successfully registered. Please verify email using link send"
+    else
+      flash[:notice] = 'User registration unsuccessful.'
+    end
+    redirect_to registration_index_path
+  end
+
+  def verification
+    debugger
+  end
+
+  private
+  def confirm_password
+    unless params[:password].empty? && params[:password] == params[:password_confirmation]
```

Comment:
Also check if password_confirmation is empty. Extract the logic in separate method